### PR TITLE
Medication & Immunization profiles refactoring slices to be additional bindings

### PR DESCRIPTION
**Pull request by dt-r**
Fixes for:
- [FHIR-44823](https://jira.hl7.org/browse/FHIR-44823) AU Base Medication profiles: Refactor to use additionalBindings instead of slices
- [FHIR-44821](https://jira.hl7.org/browse/FHIR-44821) AU Base Immunization.vaccineCode: Refactor to use additionalBindings instead of slices

Changed:
- change log
- AU Base Medication
- AU Base Medication Request
- AU Base Medication Dispense
- AU Base Medication Statement
- AU Base Medication Administration
- AU Base Immunisation

When approved DO NOT MERGE. This needs to wait for parallel AU Core change to be ready.

### CODE REVIEW

File: input/resources/au-immunization.xml [hunk 1]
```diff
@@ -26,47 +26,34 @@
...
-        <valueSet value="https://healthterminologies.gov.au/fhir/ValueSet/australian-immunisation-register-vaccine-1" />
+        <extension url="http://hl7.org/fhir/tools/StructureDefinition/additional-binding">
+          <extension url="purpose">
+            <valueCode value="preferred"/>
+          </extension>
+          <extension url="valueSet">
+            <valueCanonical value="https://healthterminologies.gov.au/fhir/ValueSet/amt-vaccine-1"/>
+          </extension>
+          <extension url="documentation">
+            <valueMarkdown value="This value set is for use when AMT (Australian Medicines Terminology) are preferred."/>
```
Comment (dbojicic) [inline input/resources/au-immunization.xml:40]: Missing word 'codes' after (Australian Medicines Terminology).

File: input/pagecontent/changes.md [hunk 2]
```diff
@@ -94,6 +94,12 @@ To help implementers, only the more significant changes are listed here.
...
+  <li>Changed MedicationDispense.medicationCodeableConcept to replace slices with Additional Bindings as 'preferred' in <a href="StructureDefinition-au-medicationdispense.html">AU Base Medication Dispense</a> (<a href="https://jira.hl7.org/browse/FHIR-44823">FHIR-44823</a>).</li>
+  <li>Changed MedicationRequest.medicationCodeableConcept to replace slices with Additional Bindings as 'preferred' in <a href="StructureDefinition-au-medicationrequest.html">AU Base Medication Request</a> (<a href="https://jira.hl7.org/browse/FHIR-44823">FHIR-44823</a>).</li>
+  <li>Changed MedicationStatement.medicationCodeableConcept to replace slices with Additional Bindings as 'preferred' in <a href="StructureDefinition-au-medicationstatement.html">AU Base Medication Statement</a> (<a href="https://jira.hl7.org/browse/FHIR-44823">FHIR-44823</a>).</li>
+  <li>Changed Immunization.code to replace slices with Additional Bindings as 'preferred' in <a href="StructureDefinition-au-medication.html">AU Base Immunisation</a> (<a href="https://jira.hl7.org/browse/FHIR-44821">FHIR-44821</a>).</li>
```
Comment (dbojicic) [inline input/pagecontent/changes.md:102]: Change hyperlink to go to StructureDefinition-au-immunization.html.

**Review by dbojicic — CHANGES_REQUESTED** [review]
A few changes needed

File: input/resources/au-medication.xml [hunk 3]
```diff
@@ -32,61 +32,56 @@
...
-      <comment value="GTIN value from http://www.gs1.org/gtin."/>
-      <binding>
-        <strength value="preferred"/>
-        <valueSet value="http://terminology.hl7.org/ValueSet/v3-GTIN"/>
+        <extension url="http://hl7.org/fhir/tools/StructureDefinition/additional-binding">
+          <extension url="purpose">
+            <valueCode value="preferred"/>
+          </extension>
+          <extension url="valueSet">
+            <valueCanonical value="https://healthterminologies.gov.au/fhir/ValueSet/australian-medication-1"/>
```
Comment (JayMurdoch) [inline input/resources/au-medication.xml:42]: This opens to https://tx.ontoserver.csiro.au/fhir/ValueSet/australian-medication-1

File: input/resources/au-medication.xml [hunk 4]
```diff
@@ -32,61 +32,56 @@
...
+        </extension>
+        <extension url="http://hl7.org/fhir/tools/StructureDefinition/additional-binding">
+          <extension url="purpose">
+            <valueCode value="preferred"/>
+          </extension>
+          <extension url="valueSet">
+            <valueCanonical value="http://terminology.hl7.org.au/ValueSet/mims"/>
+          </extension>
+          <extension url="documentation">
+            <valueMarkdown value="This value set is for use when MIMS codes are preferred."/>
```
Comment (JayMurdoch) [inline input/resources/au-medication.xml:67]: Flagging for consideration: in the table AMT, PBS and GTIN acronyms are spelled out in full, should MIMS be too? Noting that it is rarely seen in full form.

File: input/resources/au-medicationadministration.xml [hunk 5]
```diff
@@ -43,6 +43,53 @@
       <type>
         <code value="CodeableConcept"/>
       </type>
+      <binding>
+        <extension url="http://hl7.org/fhir/tools/StructureDefinition/additional-binding">
+          <extension url="purpose">
+            <valueCode value="preferred"/>
+          </extension>
+          <extension url="valueSet">
+            <valueCanonical value="https://healthterminologies.gov.au/fhir/ValueSet/australian-medication-1"/>
```
Comment (JayMurdoch) [inline input/resources/au-medicationadministration.xml:52]: This opens to https://tx.ontoserver.csiro.au/fhir/ValueSet/australian-medication-1

File: input/resources/au-medicationdispense.xml [hunk 6]
```diff
@@ -78,58 +78,55 @@
...
-      <short value="AMT Medicines"/>
       <binding>
-        <strength value="preferred"/>
-        <valueSet value="https://healthterminologies.gov.au/fhir/ValueSet/australian-medication-1"/>
+        <extension url="http://hl7.org/fhir/tools/StructureDefinition/additional-binding">
+          <extension url="purpose">
+            <valueCode value="preferred"/>
+          </extension>
+          <extension url="valueSet">
+            <valueCanonical value="https://healthterminologies.gov.au/fhir/ValueSet/australian-medication-1"/>
```
Comment (JayMurdoch) [inline input/resources/au-medicationdispense.xml:87]: This link opens to https://tx.ontoserver.csiro.au/fhir/ValueSet/australian-medication-1

File: input/resources/au-medicationrequest.xml [hunk 7]
```diff
@@ -68,58 +68,55 @@
...
-      <comment value="GTIN value from http://www.gs1.org/gtin."/>
       <binding>
-        <strength value="preferred"/>
-        <valueSet value="http://terminology.hl7.org/ValueSet/v3-GTIN"/>
+        <extension url="http://hl7.org/fhir/tools/StructureDefinition/additional-binding">
+          <extension url="purpose">
+            <valueCode value="preferred"/>
+          </extension>
+          <extension url="valueSet">
+            <valueCanonical value="https://healthterminologies.gov.au/fhir/ValueSet/australian-medication-1"/>
```
Comment (JayMurdoch) [inline input/resources/au-medicationrequest.xml:77]: This link opens to https://tx.ontoserver.csiro.au/fhir/ValueSet/australian-medication-1.

File: input/resources/au-immunization.xml [hunk 8]
```diff
@@ -26,47 +26,34 @@
...
+      <path value="Immunization.vaccineCode" />
       <binding>
-        <strength value="required" />
-        <valueSet value="https://healthterminologies.gov.au/fhir/ValueSet/australian-immunisation-register-vaccine-1" />
+        <extension url="http://hl7.org/fhir/tools/StructureDefinition/additional-binding">
+          <extension url="purpose">
+            <valueCode value="preferred"/>
+          </extension>
+          <extension url="valueSet">
+            <valueCanonical value="https://healthterminologies.gov.au/fhir/ValueSet/amt-vaccine-1"/>
```
Comment (JayMurdoch) [inline input/resources/au-immunization.xml:37]: This link opens to https://tx.ontoserver.csiro.au/fhir/ValueSet/amt-vaccine-1

File: input/resources/au-immunization.xml [hunk 9]
```diff
@@ -26,47 +26,34 @@
...
+          <extension url="documentation">
+            <valueMarkdown value="This value set is for use when AMT (Australian Medicines Terminology) are preferred."/>
+          </extension>
+        </extension>
+        <extension url="http://hl7.org/fhir/tools/StructureDefinition/additional-binding">
+          <extension url="purpose">
+            <valueCode value="preferred"/>
+          </extension>
+          <extension url="valueSet">
+            <valueCanonical value="https://healthterminologies.gov.au/fhir/ValueSet/australian-immunisation-register-vaccine-1"/>
```
Comment (JayMurdoch) [inline input/resources/au-immunization.xml:48]: This link opens to https://tx.ontoserver.csiro.au/fhir/ValueSet/australian-immunisation-register-vaccine-1

File: input/resources/au-immunization.xml (hunk 8 above)
Comment (dt-r) [reply input/resources/au-immunization.xml:37]: @JayMurdoch - that's nice. What's the review comment?

File: input/resources/au-immunization.xml (hunk 9 above)
Comment (dt-r) [reply input/resources/au-immunization.xml:48]: @JayMurdoch - that's nice. What's the review comment, i.e. what has to change and why?

File: input/resources/au-medication.xml [hunk 10]
```diff
@@ -32,61 +32,56 @@
...
-        <rules value="open"/>
-      </slicing>
-    </element>
-    <element id="Medication.code.coding:pbs">
-      <path value="Medication.code.coding"/>
-      <sliceName value="pbs"/>
-      <short value="PBS Item Code"/>
-      <comment
-        value="PBS code from http://pbs.gov.au/code/item. Use of PBS as a code to refer to a type of medication, this excludes implication of context based on the PBS code. Where context is to be implied or stated PBS code needs to be associated with recording a prescription (MedicationRequest) or dispense record (MedicationDispense)."/>
+      <comment value="Australian coding are typical medicine/product concept codes.&#xD;&#xA;&#xD;&#xA;A code (or set of codes) that specify this medication, or a textual description if no code is available. Usage note: This could be a standard medication code such as a code from RxNorm, SNOMED CT, IDMP etc. It could also be a national or local formulary code, optionally with translations to other code systems."/>
```
[the remark this answers is not in the document]
Comment (dt-r) [reply input/resources/au-medication.xml:35]: Not in scope of review. This hasn't changed.

[the remark this answers is not in the document]
Comment (dt-r) [reply input/resources/au-medication.xml:35]: @JayMurdoch - that's nice. What's the review comment, i.e. what has to change and why?

File: input/resources/au-medication.xml (hunk 3 above)
Comment (dt-r) [reply input/resources/au-medication.xml:42]: @JayMurdoch - that's nice. What's the review comment, i.e. what has to change and why?

File: input/resources/au-medication.xml (hunk 4 above)
Comment (dt-r) [reply input/resources/au-medication.xml:67]: What is MIMS an abbreviation for?

File: input/resources/au-medicationadministration.xml (hunk 5 above)
Comment (dt-r) [reply input/resources/au-medicationadministration.xml:52]: @JayMurdoch - that's nice. What's the review comment, i.e. what has to change and why?

File: input/resources/au-medicationdispense.xml (hunk 6 above)
Comment (dt-r) [reply input/resources/au-medicationdispense.xml:87]: @JayMurdoch - that's nice. What's the review comment, i.e. what has to change and why?

File: input/resources/au-medicationrequest.xml (hunk 7 above)
Comment (dt-r) [reply input/resources/au-medicationrequest.xml:77]: @JayMurdoch - that's nice. What's the review comment, i.e. what has to change and why?

File: input/resources/au-medication.xml (hunk 3 above)
Comment (JayMurdoch) [reply input/resources/au-medication.xml:42]: I now understand this is a known issue, and no change is required. Thanks.

File: input/resources/au-medicationadministration.xml (hunk 5 above)
Comment (JayMurdoch) [reply input/resources/au-medicationadministration.xml:52]: I now understand this is a known issue, and no change is required. Thanks.

File: input/resources/au-medicationdispense.xml (hunk 6 above)
Comment (JayMurdoch) [reply input/resources/au-medicationdispense.xml:87]: I now understand this is a known issue, and no change is required. Thanks.

File: input/resources/au-medicationrequest.xml (hunk 7 above)
Comment (JayMurdoch) [reply input/resources/au-medicationrequest.xml:77]: I now understand this is a known issue, and no change is required. Thanks.

File: input/resources/au-immunization.xml (hunk 8 above)
Comment (JayMurdoch) [reply input/resources/au-immunization.xml:37]: I now understand this is a known issue, and no change is required. Thanks.

File: input/resources/au-immunization.xml (hunk 9 above)
Comment (JayMurdoch) [reply input/resources/au-immunization.xml:48]: I now understand this is a known issue, and no change is required. Thanks.

File: input/resources/au-medication.xml (hunk 4 above)
Comment (JayMurdoch) [reply input/resources/au-medication.xml:67]: Suggest amending to "This value set is for use when MIMS (Monthly Index of Medical Specialties) codes are preferred."

File: input/resources/au-medicationadministration.xml (hunk 5 above)
Comment (dt-r) [reply input/resources/au-medicationadministration.xml:52]: Please resolve the conversation when you are satisfied.

File: input/resources/au-medicationdispense.xml (hunk 6 above)
Comment (dt-r) [reply input/resources/au-medicationdispense.xml:87]: Please resolve the conversation when you are satisfied.

File: input/resources/au-medicationrequest.xml (hunk 7 above)
Comment (dt-r) [reply input/resources/au-medicationrequest.xml:77]: Please resolve the conversation when you are satisfied.

File: input/resources/au-immunization.xml (hunk 8 above)
Comment (dt-r) [reply input/resources/au-immunization.xml:37]: Please resolve the conversation when you are satisfied.

File: input/resources/au-immunization.xml (hunk 9 above)
Comment (dt-r) [reply input/resources/au-immunization.xml:48]: Please resolve the conversation when you are satisfied.

File: input/resources/au-medication.xml (hunk 3 above)
Comment (JayMurdoch) [reply input/resources/au-medication.xml:42]: This comment is now resolved.

File: input/resources/au-medicationadministration.xml (hunk 5 above)
Comment (JayMurdoch) [reply input/resources/au-medicationadministration.xml:52]: This comment is now resolved.

File: input/resources/au-medicationdispense.xml (hunk 6 above)
Comment (JayMurdoch) [reply input/resources/au-medicationdispense.xml:87]: This comment is now resolved.

File: input/resources/au-medicationrequest.xml (hunk 7 above)
Comment (JayMurdoch) [reply input/resources/au-medicationrequest.xml:77]: This comment is now resolved.

File: input/resources/au-immunization.xml (hunk 8 above)
Comment (JayMurdoch) [reply input/resources/au-immunization.xml:37]: This comment is now resolved.

File: input/resources/au-immunization.xml (hunk 9 above)
Comment (JayMurdoch) [reply input/resources/au-immunization.xml:48]: This comment is now resolved.

File: input/resources/au-medication.xml (hunk 10 above)
[the remark this answers is not in the document]
Comment (dbojicic) [reply input/resources/au-medication.xml:35]: Noting that this PR has already modified the comment. While removing the word 'slices' aligns with the scope of this PR, could we also consider the change to make the resulting content grammatically correct to be in scope?

[the remark this answers is not in the document]
Comment (dt-r) [reply input/resources/au-medication.xml:35]: It's not in scope but I'll change it anyway.

File: input/resources/au-immunization.xml (hunk 1 above)
Comment (dt-r) [reply input/resources/au-immunization.xml:40]: fixed.

File: input/pagecontent/changes.md (hunk 2 above)
Comment (dt-r) [reply input/pagecontent/changes.md:102]: fixed.

File: input/resources/au-medication.xml (hunk 10 above)
[the remark this answers is not in the document]
Comment (dt-r) [reply input/resources/au-medication.xml:35]: fixed.

File: input/resources/au-medication.xml (hunk 4 above)
Comment (dt-r) [reply input/resources/au-medication.xml:67]: fixed.

File: input/resources/au-immunization.xml (hunk 1 above)
Comment (dbojicic) [reply input/resources/au-immunization.xml]: Confirmed fix.

File: input/pagecontent/changes.md (hunk 2 above)
Comment (dbojicic) [reply input/pagecontent/changes.md]: Confirmed fix.

File: input/resources/au-medication.xml (hunk 10 above)
[the remark this answers is not in the document]
Comment (dbojicic) [reply input/resources/au-medication.xml]: Confirmed fix.

**Review by dbojicic — APPROVED** [review]

**Review by JayMurdoch — APPROVED** [review]
Changes reviewed, no further issues.

**Pull request merged**